core/src/tests/test-parachain/src/xcm.rs

```diff
@@ -115,16 +118,108 @@ pub type XcmRouter = (
 
 /////////// Other pallets
 
+parameter_types! {
+	pub const RelayChain: Location = Location::parent();
+
+	/// The asset ID for the asset that we use to pay for message delivery fees.
+	pub FeeAssetId: AssetId = AssetId(RelayChain::get());
+	/// The base fee for the message delivery fees.
+	pub const BaseDeliveryFee: Balance = 300_000_000;
+	/// The fee per byte
+	pub const ByteFee: Balance = 1_000_000;
+}
+
+pub type PriceForSiblingParachainDelivery = polkadot_runtime_common::xcm_sender::ExponentialPrice<
+	FeeAssetId,
+	BaseDeliveryFee,
+	ByteFee,
+	XcmpQueue,
+>;
+
+parameter_types! {
+	pub const HeapSize: u32 = 1024;
+	pub const MaxStale: u32 = 2;
+	pub const ServiceWeight: Option<Weight> = Some(Weight::from_parts(100, 100));
+}
+
+pub struct MessageQueueWeightInfo;
+
+// As of this version, the current implementation of WeightInfo for (), has weights that are bigger than 0.
+impl pallet_message_queue::WeightInfo for MessageQueueWeightInfo {
+	fn ready_ring_knit() -> Weight {
+		Weight::zero()
+	}
+
+	fn ready_ring_unknit() -> Weight {
+		Weight::zero()
+	}
+
+	fn service_queue_base() -> Weight {
+		Weight::zero()
+	}
+
+	fn service_page_base_completion() -> Weight {
+		Weight::zero()
+	}
+
+	fn service_page_base_no_completion() -> Weight {
+		Weight::zero()
+	}
+
+	fn service_page_item() -> Weight {
+		Weight::zero()
+	}
+
+	fn bump_service_head() -> Weight {
+		Weight::zero()
+	}
+
+	fn reap_page() -> Weight {
+		Weight::zero()
+	}
+
+	fn execute_overweight_page_removed() -> Weight {
+		Weight::zero()
+	}
+
+	fn execute_overweight_page_updated() -> Weight {
+		Weight::zero()
+	}
+}
+
+impl pallet_message_queue::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = MessageQueueWeightInfo;
+	type MessageProcessor = xcm_builder::ProcessXcmMessage<
+		AggregateMessageOrigin,
+		xcm_executor::XcmExecutor<XcmConfig>,
+		RuntimeCall,
+	>;
+	type Size = u32;
+	type QueueChangeHandler = ();
+	type QueuePausedQuery = ();
+	type HeapSize = HeapSize;
+	type MaxStale = MaxStale;
+	type ServiceWeight = ServiceWeight;
+}
+
+pub struct ParaIdToSibling;
+impl sp_runtime::traits::Convert<ParaId, AggregateMessageOrigin> for ParaIdToSibling {
+	fn convert(para_id: ParaId) -> AggregateMessageOrigin {
+		AggregateMessageOrigin::Sibling(para_id)
+	}
+}
+
 impl cumulus_pallet_xcmp_queue::Config for Runtime {
 	type ChannelInfo = ParachainSystem;
 	type ControllerOrigin = EnsureRoot<AccountId>;
 	type ControllerOriginConverter = ();
-	type ExecuteOverweightOrigin = EnsureRoot<AccountId>;
-	type PriceForSiblingDelivery = ();
+	type MaxInboundSuspended = ();
```

Comment:
This should be initialized with some positive number as it needs to be larger than max number of channels according to the [MQ PR](https://github.com/paritytech/polkadot-sdk/pull/1246)
```suggestion
	type MaxInboundSuspended = Const<1000u64>;
```

Reply:
Good eye @wischli! They seemed to be the changes we needed to make in the correct direction to make this work.